### DISCUSSION TITLE
关于web代理和302转发的问题

### DISCUSSION THREAD

**Discussion by waystac** [AlistGo/alist#2342] [Invalid question]
我用的是v3版本的alist，一个是在电脑挂载的，一个是在手机用Termux挂载的，还有一个是在电视盒子用Termux挂载的，都存在以下几个疑点。

1 如果不打开web代理，无法通过网页端读取文件（可以查看目录）。
2 如果使用302转发，则无法使用webdav读取文件（可以查看目录），需要改成本地代理才能读取得到（试过reex、kodi、x-plore都不行）。但是我在用别人，比如dubi.tk的资源时，他是使用302转发，我可以成功读取文件的（我不知道dubi.tk用的是什么版本的alist）。

想请问一下这是为什么，应该怎么解决这个问题呢？谢谢。

**Comment by sha7dowXX** [comment]
同问

**Comment by waystac** [comment]
找到原因了，一定要按https://alist.nn.ci/zh/guide/drivers/aliyundrive.html提供的方式取得refresh token，才能在302转发时成功读取文件。另外要保证客户端支持302转发。

**Reply by sha7dowXX** [reply]
> 找到原因了，一定要按[https://alist.nn.ci/zh/guide/drivers/aliyundrive.html提供的方式取得refresh](https://alist.nn.ci/zh/guide/drivers/aliyundrive.html%E6%8F%90%E4%BE%9B%E7%9A%84%E6%96%B9%E5%BC%8F%E5%8F%96%E5%BE%97refresh) token，才能在302转发时成功读取文件。另外要保证客户端支持302转发。

恭喜，可是我是onedrive出现这问题的……裂开

**Reply by Java2Boy** [reply]
遇到同样问题，sharepoint.com可以Ping通，但网页端无法在线播放文件

**Comment by oneisall8955** [comment]
按照文档，需要用app的的refresh token
